### PR TITLE
Update deploying.md

### LINKED ISSUE #1351: Proofread, clarify, and update Deploying with Heroku outline
Proofread, clarify, and update Deploying with Heroku outline:
https://github.com/Techtonica/curriculum/blob/master/deploying/deploying.md

Please make sure:
- [ ] The language is as clear as possible for someone who is deploying a site to heroku for the first time, and deploying a full-stack app for the first time ever.
- [ ] Make sure all Heroku directions and resources are not out-of-date -- replace them if they are. either way, feel free to replace with better ones.
- [ ] Make sure any updates are also applied to the slideshow linked in the outline. (Request edit access)
- [ ]  It could be a lot more relevant than it is now - consider skipping info about containers, etc if it's not immediately useful to someone deploying a small simple React App.

### PR DESCRIPTION
**Pull request by kichloo**
> 
> * [X]  The language is as clear as possible for someone who is deploying a site to heroku for the first time, and deploying a full-stack app for the first time ever.
> * [X]  Make sure all Heroku directions and resources are not out-of-date -- replace them if they are.
> * [X]  Make sure any updates are also applied to the slideshow linked in the outline.

Closes #1351

### CODE REVIEW

File: deploying/deploying.md [hunk 1]
```diff
@@ -45,7 +45,7 @@ Deployment is a fancy term for getting your website on the web. After building o
 
 Now we'll work on deploying your app to Heroku.
 
-We'll be combining your frontend (create-react-app) with your backend (express) and deploying it to Heroku. Your frontend contains "static" JavaScript files -- when you deploy to Heroku, Heroku turns your whole React app into a couple of static files that it will serve to the browser. No matter what data you have in the database, these files will always be the same.
+We'll be combining your frontend (create-react-app) with your backend (express) and deploying it to Heroku. Your frontend contains "static" JavaScript files -- when you deploy to Heroku, it turns your whole React app into a couple of static files that it will serve to the browser. No matter what data you have in the database, these files will always be the same.
```
Comment (Kundan28) [inline deploying/deploying.md:48]:
```suggestion
We'll be combining your front-end(create-react-app) with your back-end (express) and deploying it to Heroku. Your front-end contains "static" JavaScript files -- when you deploy to Heroku, it turns your whole React app into a couple of static files that it will serve to the browser. No matter what data you have in the database, these files will always be the same.
```

File: deploying/deploying.md [hunk 2]
```diff
@@ -171,7 +171,7 @@ Ensure you don't have any missing files: `git status` and commit them if you nee
 13. Deploy your app!
     `git push heroku master`
     This takes a long time.
-    This will print the URL your app was deployed to. Trying going to it! If something goes run, use `heroku logs --tail` to debug.
+    This will print the URL your app was deployed to. Try going to it! If something goes run, use `heroku logs --tail` to debug.
```
Comment (Kundan28) [inline deploying/deploying.md:174]:
```suggestion
    This will print the URL your app was deployed to. Try going to it! If something goes wrong, use `heroku logs --tail` to debug.
```

File: deploying/deploying.md [hunk 3]
```diff
@@ -193,4 +193,4 @@ All done! Small differences in the way you've set up your site may make bits of
 
 - [Tutorial - Heroku Dev Center Deployment](https://devcenter.heroku.com/articles/deploying-nodejs)
 - [Overview of Deployment Options - MDN Express & Node Deployment](https://developer.mozilla.org/en-US/docs/Learn/Server-side/Express_Nodejs/deployment)
-- [Code School Tutorial(requires sign-in) - Domain Names,Name Servers, and Cloud Based Servers](https://www.codeschool.com/beginners-guide-to-web-development/deploying-your-first-website)
+- [Code School Tutorial(requires sign-in) - Domain Names,Name Servers and Cloud Based Servers](https://www.codeschool.com/beginners-guide-to-web-development/deploying-your-first-website)
```
Comment (Kundan28) [inline deploying/deploying.md:196]:
```suggestion
- [Code School Tutorial(requires sign-in) - Domain Names, Name Servers and Cloud Based Servers](https://www.codeschool.com/beginners-guide-to-web-development/deploying-your-first-website)
```

Comment (Kundan28) [inline deploying/deploying.md:196]: This link is redirecting to `https://www.pluralsight.com/paths/html5` so better to switch it with that or find other one.

**Comment by kichloo** [comment]
@Kundan28 Done

File: deploying/deploying.md [hunk 4]
```diff
@@ -193,4 +193,4 @@ All done! Small differences in the way you've set up your site may make bits of
 
 - [Tutorial - Heroku Dev Center Deployment](https://devcenter.heroku.com/articles/deploying-nodejs)
 - [Overview of Deployment Options - MDN Express & Node Deployment](https://developer.mozilla.org/en-US/docs/Learn/Server-side/Express_Nodejs/deployment)
-- [Code School Tutorial(requires sign-in) - Domain Names,Name Servers and Cloud Based Servers](https://www.codeschool.com/beginners-guide-to-web-development/deploying-your-first-website)
+- [Code School Tutorial(requires sign-in) - Domain Names, Name Servers and Cloud Based Servers](https://www.pluralsight.com/paths/html5)
```
Comment (Kundan28) [inline deploying/deploying.md:196]: You have only changed the link but forgot to change the Label.

Comment (kichloo) [reply deploying/deploying.md:196]: Can you suggest a proper label name for it?

Comment (Kundan28) [reply deploying/deploying.md:196]: `HTML5`. And change the description too.

Comment (alodahl) [reply deploying/deploying.md:196]: Please only provide free courses.  I think this one requires a membership, plus I don't think HTML5 it is a good replacement for  a tutorial about domain names and servers.

**Comment by alodahl** [comment]
Hi @kichloo , could you please follow the tutorial yourself and then fix any problems you find with it?

File: deploying/deploying.md (hunk 4 above)
Comment (kichloo) [reply deploying/deploying.md:196]: Reversed the changes again

Comment (kichloo) [reply deploying/deploying.md:196]: @alodahl The link was of HTML5 only!

**Comment by kichloo** [comment]
@alodahl I had gone through the ` .md` file and done the required changes and submitted the PR. But it was not mentioned that the courses have to be free. I will fix those.

File: deploying/deploying.md (hunk 4 above)
Comment (alodahl) [reply deploying/deploying.md:196]: Yes, it is wrong and outdated.  Please replace it with a comparable tutorial or delete the whole line.

**Comment by alodahl** [comment]
@kichloo do you have any opinions from following the outline?  
Was it easy to understand?
Were any of the directions (video, blog, or in this outline) outdated compared to how Heroku works today?

File: deploying/deploying.md [hunk 5]
```diff
@@ -32,7 +32,7 @@ Deployment is a fancy term for getting your website on the web. After building o
 
 - [Video - What is Heroku (4 mins watch)](https://youtu.be/r5ZUQvl9BtE)
 - [Techtonica Slides: Deploying](https://docs.google.com/presentation/d/1Enwhd9hl1fn1-afMXJ6xvkJm5SDJpHjfQoA7s2znHpw/edit?usp=sharing)
-- [Deploy Node.JS Apps to Heroku (10 min Video)](https://youtu.be/AZNFox2CvBk)
+- [Deploy Node.JS Apps to Heroku (10 mins Video)](https://youtu.be/AZNFox2CvBk)
```
Comment (Aayush-1999) [inline deploying/deploying.md:35]: I think it is better to add one more material on `how to run front-end and back-end on a same server`.

File: deploying/deploying.md [hunk 6]
```diff
@@ -45,7 +45,7 @@ Deployment is a fancy term for getting your website on the web. After building o
...
 
-We'll be combining your frontend (create-react-app) with your backend (express) and deploying it to Heroku. Your frontend contains "static" JavaScript files -- when you deploy to Heroku, Heroku turns your whole React app into a couple of static files that it will serve to the browser. No matter what data you have in the database, these files will always be the same.
+We'll be combining your front-end (create-react-app) with your back-end (express) and deploying it to Heroku. Your front-end contains "static" JavaScript files -- when you deploy to Heroku, it turns your whole React app into a couple of static files that it will serve to the browser. No matter what data you have in the database, these files will always be the same.
 
 Your backend, on the other hand, is dynamic -- when you make an API request, the backend runs javascript code to do things like reading and writing to a database. Unlike the React app, which always serves the same files to the browser, the backend will serve different information to the browser depending on what's in the database. We're going to combine your dynamic code (express), with your static code (create-react-app).
```
Comment (Aayush-1999) [inline deploying/deploying.md:50]: After this, the first point should be **cd into your app and create 2 new directories `client` and `server`**. The reason for this is we don't now what kind of directory structure they have followed ( like React-app inside the Back-end or vice-versa). It will be more better if you add a image of the final directory structure after 2nd point.

File: deploying/deploying.md (hunk 5 above)
Comment (kichloo) [reply deploying/deploying.md:35]: I found this one: [https://www.twilio.com/blog/react-app-with-node-js-server-proxy]
Do tell if you want me to include this or you have a better resource

File: deploying/deploying.md [hunk 7]
```diff
@@ -127,8 +127,14 @@ file.
     Through the Heroku web UI, create a new Application.
     Once you create the app, add the Postgres add-on by going to the Resources tab
     and searching in the "Add-ons" search box for Postgres.
-    Click the "Heroku Postgres" option. Finally, select the free version and click
-    "Provision".
+    Click the "Heroku Postgres" option. Finally, select the free version and click "Provision". 
+    Or you can create the new Application fom the CLI. To create an application with name,install the Heroku CLI and run the following command
```
Comment (Kundan28) [inline deploying/deploying.md:131]:
```suggestion
    Or you can create the new Application from the CLI. To create an application with name, install the Heroku CLI and run the following command
```

File: deploying/deploying.md [hunk 8]
```diff
@@ -33,6 +33,7 @@ Deployment is a fancy term for getting your website on the web. After building o
 - [Video - What is Heroku (4 mins watch)](https://youtu.be/r5ZUQvl9BtE)
 - [Techtonica Slides: Deploying](https://docs.google.com/presentation/d/1Enwhd9hl1fn1-afMXJ6xvkJm5SDJpHjfQoA7s2znHpw/edit?usp=sharing)
 - [Deploy Node.JS Apps to Heroku (10 mins Video)](https://youtu.be/AZNFox2CvBk)
+- []
```
Comment (Kundan28) [inline deploying/deploying.md]: Why you added this ?

File: deploying/deploying.md [hunk 9]
```diff
@@ -49,7 +50,7 @@ We'll be combining your front-end (create-react-app) with your back-end (express
 
 Your backend, on the other hand, is dynamic -- when you make an API request, the backend runs javascript code to do things like reading and writing to a database. Unlike the React app, which always serves the same files to the browser, the backend will serve different information to the browser depending on what's in the database. We're going to combine your dynamic code (express), with your static code (create-react-app).
 
-1. cd into the React app you created and move _everything_ into a new directory named `client`:
+1. cd into the React app and create 2 new directories `client` and `server` you created and move _everything_ into the  new directory `client`:
```
Comment (Kundan28) [inline deploying/deploying.md:52]:
```suggestion
1. cd into the React app and create 2 new directories `client` and `server` you created and move _everything_ into the new directory `client`:
```

File: deploying/deploying.md (hunk 6 above)
Comment (alodahl) [reply deploying/deploying.md:50]: Hi @Aayush-1999 , it isn't part of the issue to fix this, but thanks for pointing it out.  I will make sure it gets made into a new issue.

File: deploying/deploying.md (hunk 5 above)
Comment (alodahl) [reply deploying/deploying.md:35]: I don't understand why - this outline is specifically about heroku.

File: deploying/deploying.md [hunk 10]
```diff
@@ -45,11 +45,11 @@ Deployment is a fancy term for getting your website on the web. After building o
...
+We'll be combining your front-end (create-react-app) with your back-end (express) and deploying it to Heroku. Your front-end contains "static" JavaScript files -- when you deploy to Heroku, it turns your whole React app into a couple of static files that it will serve to the browser. No matter what data you have in the database, these files will always be the same.
 
 Your backend, on the other hand, is dynamic -- when you make an API request, the backend runs javascript code to do things like reading and writing to a database. Unlike the React app, which always serves the same files to the browser, the backend will serve different information to the browser depending on what's in the database. We're going to combine your dynamic code (express), with your static code (create-react-app).
 
-1. cd into the React app you created and move _everything_ into a new directory named `client`:
+1. cd into the React app and create 2 new directories `client` and `server` you created and move _everything_ into the new directory `client`:
```
Comment (alodahl) [inline deploying/deploying.md:52]: Thank you for adding that, but could you rephrase the sentence now?  It doesn't make sense anymore.

File: deploying/deploying.md [hunk 11]
```diff
@@ -117,7 +117,7 @@ This block of code only runs in production. When it runs, it will serve your Jav
 
 ```json
     "start": "node server/server.js",
-    "heroku-postbuild": "cd client && npm install && npm install --only=dev --no-shrinkwrap && npm run build"
+    "heroku-postbuild": "npm install --only=dev --no-shrinkwrap && npm run build"
```
Comment (alodahl) [inline deploying/deploying.md:120]: 💜

File: deploying/deploying.md [hunk 12]
```diff
@@ -193,4 +196,3 @@ All done! Small differences in the way you've set up your site may make bits of
 
 - [Tutorial - Heroku Dev Center Deployment](https://devcenter.heroku.com/articles/deploying-nodejs)
 - [Overview of Deployment Options - MDN Express & Node Deployment](https://developer.mozilla.org/en-US/docs/Learn/Server-side/Express_Nodejs/deployment)
-- [Code School Tutorial(requires sign-in) - Domain Names,Name Servers, and Cloud Based Servers](https://www.codeschool.com/beginners-guide-to-web-development/deploying-your-first-website)
```
Comment (alodahl) [inline deploying/deploying.md:196]: 👍🏼

**Comment by alodahl** [comment]
Almost there!  Thanks @kichloo !

File: deploying/deploying.md (hunk 5 above)
Comment (kichloo) [reply deploying/deploying.md:35]: Ok, I won't add it.

**Comment by kichloo** [comment]
Did the required changes.

**Comment by alodahl** [comment]
Were you able to make the changes @Aayush-1999 requested here? https://github.com/Techtonica/curriculum/pull/1356#pullrequestreview-413172134

**Comment by kichloo** [comment]
Yes, did them but the link which mentioned was discarded, so give a proper link for it.

File: deploying/deploying.md [hunk 13]
```diff
@@ -127,8 +127,14 @@ file.
     Through the Heroku web UI, create a new Application.
     Once you create the app, add the Postgres add-on by going to the Resources tab
     and searching in the "Add-ons" search box for Postgres.
-    Click the "Heroku Postgres" option. Finally, select the free version and click
-    "Provision".
+    Click the "Heroku Postgres" option. Finally, select the free version and click "Provision". 
+    Or you can create the new Application from the CLI. To create an application with name, install the Heroku CLI and run the following command
```
Comment (Kundan28) [inline deploying/deploying.md:131]:
```suggestion
    Or you can create a new Application from the CLI. To create an application with name, install the Heroku CLI and run the following command
```

**Comment by kichloo** [comment]
Done

**Comment by Kundan28** [comment]
> Done

Resolve the conflicts.

**Comment by kichloo** [comment]
> Resolve the conflicts.

Done

**Comment by Kundan28** [comment]
> Done

Now wait for the other mentors to review it.

**Comment by Kundan28** [comment]
Updated the scoresheet too,
